On Pilot Logging

There are a few issues with the current implementation of PilotLogging. 

1) **SOLVED** activating *RemoteLogger* in an environment that does NOT include env `X509_CERT_DIR` and `X509_USER_PROXY` generates the following:

```
2023-10-31T15:48:30.341093Z DEBUG [PilotParams] Release project: 
2023-10-[31](https://github.com/fstagni/Pilot/actions/runs/6709010578/job/18231106705#step:6:32)T15:48:30.342023Z INFO [Pilot] Remote logger activated
Traceback (most recent call last):
  File "/home/runner/work/Pilot/Pilot/Pilot/dirac-pilot.py", line 76, in <module>
    log.buffer.flush()
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 514, in wrapper
    return func(self, *args, **kwargs)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 599, in flush
    self.senderFunc(buf)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 6[35](https://github.com/fstagni/Pilot/actions/runs/6709010578/job/18231106705#step:6:36), in sendMessage
    context.load_cert_chain(cert)
TypeError: certfile should be a valid filesystem path
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 522, in run
    self.function(*self.args, **self.kwargs)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 514, in wrapper
    return func(self, *args, **kwargs)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 599, in flush
    self.senderFunc(buf)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 635, in sendMessage
    context.load_cert_chain(cert)
TypeError: certfile should be a valid filesystem path
```

This is especially true for the case when DIRAC is sourced from CVMFS (not installed).

To address this issue, in https://github.com/DIRACGrid/Pilot/pull/218 I moved the setting of these variables as early as possible. 


2) **NOT SOLVED** Testing the above PR (#218) when DIRAC is sourced from CVMFS, and we're using certificates and not proxies generates the following (https://github.com/fstagni/Pilot/actions/runs/6719269671/job/18260562881):

```
2023-11-01T11:50:36.304812Z DEBUG [PilotParams] Release project: 
2023-11-01T11:50:44.286345Z DEBUG [PilotParams] X509_CERT_DIR is set in the host environment as /cvmfs/grid.cern.ch/etc/grid-security/certificates, aligning installEnv to it
2023-11-01T11:50:44.287539Z DEBUG [PilotParams] X509_VOMS_DIR is set in the host environment as /cvmfs/grid.cern.ch/etc/grid-security/vomsdir, aligning installEnv to it
2023-11-01T11:50:44.287626Z DEBUG [PilotParams] X509_VOMSES is not set in the host environment
2023-11-01T11:50:44.287688Z DEBUG [PilotParams] Candidate directory for X509_VOMSES is /cvmfs/grid.cern.ch/etc/grid-security/vomses
2023-11-01T11:50:44.289156Z DEBUG [PilotParams] Setting X509_VOMSES=/cvmfs/grid.cern.ch/etc/grid-security/vomses
2023-11-01T11:50:44.289448Z INFO [Pilot] Remote logger activated
Traceback (most recent call last):
  File "/home/runner/work/Pilot/Pilot/Pilot/dirac-pilot.py", line 76, in <module>
    log.buffer.flush()
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 503, in wrapper
    return func(self, *args, **kwargs)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 588, in flush
    self.senderFunc(buf)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 6[31](https://github.com/fstagni/Pilot/actions/runs/6719269671/job/18260562881#step:6:32), in sendMessage
    res = urlopen(url, data, context=context)
  File "/usr/lib/python3.10/urllib/request.py", line 216, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.10/urllib/request.py", line 525, in open
    response = meth(req, response)
  File "/usr/lib/python3.10/urllib/request.py", line 6[34](https://github.com/fstagni/Pilot/actions/runs/6719269671/job/18260562881#step:6:35), in http_response
    response = self.parent.error(
  File "/usr/lib/python3.10/urllib/request.py", line 563, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.10/urllib/request.py", line 496, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.10/urllib/request.py", line 643, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 401: Unauthorized
Exception in thread Thread-4:
Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 511, in run
    self.function(*self.args, **self.kwargs)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 503, in wrapper
    return func(self, *args, **kwargs)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 588, in flush
    self.senderFunc(buf)
  File "/home/runner/work/Pilot/Pilot/Pilot/pilotTools.py", line 631, in sendMessage
    res = urlopen(url, data, context=context)
  File "/usr/lib/python3.10/urllib/request.py", line 216, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.10/urllib/request.py", line 525, in open
    response = meth(req, response)
  File "/usr/lib/python3.10/urllib/request.py", line 634, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.10/urllib/request.py", line 563, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.10/urllib/request.py", line 496, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.10/urllib/request.py", line 643, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error [40](https://github.com/fstagni/Pilot/actions/runs/6719269671/job/18260562881#step:6:41)1: Unauthorized
Error: Process completed with exit code 1.
```

this is because the host certificate is not recognized as such. A DIRAC client would add a `{""extraCredentials": "hosts"}` to the credential dictionary, but this is not done when running before DIRAC is installed.


3) **NOT SOLVED**. Testing the above PR (#218) when DIRAC is installed locally, and we're using certificates and not proxies generates the following (https://github.com/fstagni/Pilot/actions/runs/6719269671/job/18260566428) : 

```
2023-11-01T11:51:13.939924Z INFO [Pilot] Remote logger activated
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/__w/Pilot/Pilot/Pilot/pilotTools.py", line 511, in run
    self.function(*self.args, **self.kwargs)
  File "/__w/Pilot/Pilot/Pilot/pilotTools.py", line 503, in wrapper
    return func(self, *args, **kwargs)
  File "/__w/Pilot/Pilot/Pilot/pilotTools.py", line 588, in flush
    self.senderFunc(buf)
  File "/__w/Pilot/Pilot/Pilot/pilotTools.py", line 623, in sendMessage
    context.load_verify_locations(capath=caPath)
TypeError: cafile, capath and cadata cannot be all omitted
```

Again, this is because `X509_CERT_DIR` and `X509_USER_PROXY` are not defined at this stage. 

---

I don't know how to solve 2) and 3). 

Maybe for 2) the simplest would be to use a `diracx` service.

--- a/.github/workflows/integration.yml
+++ b/.github/workflows/integration.yml
@@ -8,7 +8,7 @@ jobs:
 
     strategy:
       matrix:
-        python: 
+        python_version: 
           - 2.7.18
           - 3.6.15
           - 3.9.17
@@ -22,10 +22,19 @@ jobs:
           - dteam
           - gridpp
 
-    container: python:${{ matrix.python }}-slim
-
     steps:
       - uses: actions/checkout@v4
+      - uses: cvmfs-contrib/github-action-cvmfs@v3
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python_version }}
+
+      - name: Check Python version
+        run: python --version
+
+      - name: Test CernVM-FS
+        run: ls /cvmfs/dirac.egi.eu
+
       - name: Retrieve the secret and decode it to a file
         env:
           HOSTCERT_BASE64: ${{ secrets.HOSTCERT_BASE64 }}
@@ -148,10 +157,19 @@ jobs:
           - pilot_oldSchema.json
           - pilot_newSchema.json
 
-    container: python:${{ matrix.python }}-slim
-    
     steps:
       - uses: actions/checkout@v4
+      - uses: cvmfs-contrib/github-action-cvmfs@v3
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python_version }}
+
+      - name: Check Python version
+        run: python --version
+
+      - name: Test CernVM-FS
+        run: ls /cvmfs/dirac.egi.eu
+
       - name: Retrieve the secret and decode it to a file
         env:
           HOSTCERT_BASE64: ${{ secrets.HOSTCERT_BASE64 }}

--- a/.github/workflows/integration.yml
+++ b/.github/workflows/integration.yml
@@ -8,10 +8,6 @@ jobs:
 
     strategy:
       matrix:
-        python_version: 
-          - 2.7.18
-          - 3.6.15
-          - 3.9.17
         dirac_version:
           - rel-v8r0
           - integration
@@ -25,12 +21,6 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - uses: cvmfs-contrib/github-action-cvmfs@v3
-      - uses: actions/setup-python@v2
-        with:
-          python-version: ${{ matrix.python_version }}
-
-      - name: Check Python version
-        run: python --version
 
       - name: Test CernVM-FS
         run: ls /cvmfs/dirac.egi.eu
@@ -158,10 +148,6 @@ jobs:
 
     strategy:
       matrix:
-        python: 
-          - 2.7.18
-          - 3.6.15
-          - 3.9.17
         pilot_schema:
           - pilot_oldSchema.json
           - pilot_newSchema.json
@@ -169,12 +155,6 @@ jobs:
     steps:
       - uses: actions/checkout@v4
       - uses: cvmfs-contrib/github-action-cvmfs@v3
-      - uses: actions/setup-python@v2
-        with:
-          python-version: ${{ matrix.python_version }}
-
-      - name: Check Python version
-        run: python --version
 
       - name: Test CernVM-FS
         run: ls /cvmfs/dirac.egi.eu

--- a/.github/workflows/integration.yml
+++ b/.github/workflows/integration.yml
@@ -18,6 +18,9 @@ jobs:
         pilot_schema:
           - pilot_oldSchema.json
           - pilot_newSchema.json
+        VO: 
+          - dteam
+          - gridpp
 
     container: python:${{ matrix.python }}-slim
 
@@ -45,7 +48,7 @@ jobs:
           sed -i "s#VAR_CS#https://lbcertifdirac70.cern.ch:9135/Configuration/Server#g" pilot.json
           sed -i "s#VAR_USERDN#/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=alboyer/CN=819281/CN=Alexandre Franck Boyer#g" pilot.json
           sed -i "s#VAR_USERDN_GRIDPP#${DIRACUSERDN_GRIDPP}#g" pilot.json
-          python dirac-pilot.py --modules https://github.com/DIRACGrid/DIRAC.git:::DIRAC:::${{ matrix.dirac_version }} -M 1 -S DIRAC-Certification -N jenkins.cern.ch -Q jenkins-queue_not_important -n DIRAC.Jenkins.ch --cert --certLocation=${GITHUB_WORKSPACE}/Pilot/etc/grid-security --debug
+          python dirac-pilot.py --modules https://github.com/DIRACGrid/DIRAC.git:::DIRAC:::${{ matrix.dirac_version }} -M 1 -S DIRAC-Certification -N jenkins.cern.ch -Q jenkins-queue_not_important -n DIRAC.Jenkins.ch --cert --certLocation=${GITHUB_WORKSPACE}/Pilot/etc/grid-security --wnVO=${{ matrix.VO }} --debug
 
 
   integration-cvmfs:

--- a/.github/workflows/integration.yml
+++ b/.github/workflows/integration.yml
@@ -57,7 +57,10 @@ jobs:
           sed -i "s#VAR_CS#https://lbcertifdirac70.cern.ch:9135/Configuration/Server#g" pilot.json
           sed -i "s#VAR_USERDN#/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=alboyer/CN=819281/CN=Alexandre Franck Boyer#g" pilot.json
           sed -i "s#VAR_USERDN_GRIDPP#${DIRACUSERDN_GRIDPP}#g" pilot.json
-          python dirac-pilot.py --modules https://github.com/DIRACGrid/DIRAC.git:::DIRAC:::${{ matrix.dirac_version }} -M 1 -S DIRAC-Certification -N jenkins.cern.ch -Q jenkins-queue_not_important -n DIRAC.Jenkins.ch --cert --certLocation=${GITHUB_WORKSPACE}/Pilot/etc/grid-security --wnVO=${{ matrix.VO }} --debug
+          g_job="testintegrationworkflow${GITHUB_JOB//-/}"
+          pilotUUID="${g_job//_/}""${GITHUB_RUN_ID}"
+          pilotUUID=$(echo $pilotUUID | rev | cut -c 1-32 | rev)
+          python dirac-pilot.py --modules https://github.com/DIRACGrid/DIRAC.git:::DIRAC:::${{ matrix.dirac_version }} -M 1 -S DIRAC-Certification -N jenkins.cern.ch -Q jenkins-queue_not_important -n DIRAC.Jenkins.ch --cert --certLocation=${GITHUB_WORKSPACE}/Pilot/etc/grid-security --wnVO=${{ matrix.VO }} --pilotUUID="${pilotUUID}" --debug
 
 
   integration-cvmfs:
@@ -105,7 +108,10 @@ jobs:
           sed -i "s#VAR_CS#https://lbcertifdirac70.cern.ch:9135/Configuration/Server#g" pilot.json
           sed -i "s#VAR_USERDN#/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=alboyer/CN=819281/CN=Alexandre Franck Boyer#g" pilot.json
           sed -i "s#VAR_USERDN_GRIDPP#${DIRACUSERDN_GRIDPP}#g" pilot.json
-          python dirac-pilot.py -M 1 -S DIRAC-Certification -N jenkins.cern.ch -Q jenkins-queue_not_important -n DIRAC.Jenkins.ch --preinstalledEnvPrefix=/cvmfs/dirac.egi.eu/dirac --cert --certLocation=${GITHUB_WORKSPACE}/Pilot/etc/grid-security --wnVO=dteam --debug
+          g_job="testintegrationworkflow${GITHUB_JOB//-/}"
+          pilotUUID="${g_job//_/}""${GITHUB_RUN_ID}"
+          pilotUUID=$(echo $pilotUUID | rev | cut -c 1-32 | rev)
+          python dirac-pilot.py -M 1 -S DIRAC-Certification -N jenkins.cern.ch -Q jenkins-queue_not_important -n DIRAC.Jenkins.ch --preinstalledEnvPrefix=/cvmfs/dirac.egi.eu/dirac --cert --certLocation=${GITHUB_WORKSPACE}/Pilot/etc/grid-security --wnVO=dteam --pilotUUID="${pilotUUID}" --debug
 
 
   integration-cvmfs_no_env:
@@ -140,7 +146,10 @@ jobs:
           sed -i "s#VAR_CS#https://lbcertifdirac70.cern.ch:9135/Configuration/Server#g" pilot.json
           sed -i "s#VAR_USERDN#/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=alboyer/CN=819281/CN=Alexandre Franck Boyer#g" pilot.json
           sed -i "s#VAR_USERDN_GRIDPP#${DIRACUSERDN_GRIDPP}#g" pilot.json
-          python dirac-pilot.py -M 1 -S DIRAC-Certification -N jenkins.cern.ch -Q jenkins-queue_not_important -n DIRAC.Jenkins.ch --preinstalledEnvPrefix=/cvmfs/dirac.egi.eu/dirac --cert --certLocation=${GITHUB_WORKSPACE}/Pilot/etc/grid-security --wnVO=dteam -o cvmfsOnly --debug
+          g_job="testintegrationworkflow${GITHUB_JOB//-/}"
+          pilotUUID="${g_job//_/}""${GITHUB_RUN_ID}"
+          pilotUUID=$(echo $pilotUUID | rev | cut -c 1-32 | rev)
+          python dirac-pilot.py -M 1 -S DIRAC-Certification -N jenkins.cern.ch -Q jenkins-queue_not_important -n DIRAC.Jenkins.ch --preinstalledEnvPrefix=/cvmfs/dirac.egi.eu/dirac --cert --certLocation=${GITHUB_WORKSPACE}/Pilot/etc/grid-security --wnVO=dteam -o cvmfsOnly --pilotUUID="${pilotUUID}" --debug
 
 
 
@@ -192,7 +201,10 @@ jobs:
           sed -i "s#VAR_CS#https://lbcertifdirac70.cern.ch:9135/Configuration/Server#g" pilot.json
           sed -i "s#VAR_USERDN#/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=alboyer/CN=819281/CN=Alexandre Franck Boyer#g" pilot.json
           sed -i "s#VAR_USERDN_GRIDPP#${DIRACUSERDN_GRIDPP}#g" pilot.json
-          python dirac-pilot.py --modules https://github.com/DIRACGrid/DIRAC.git:::DIRAC:::integration -M 1 -S DIRAC-Certification -N jenkins-full.cern.ch -Q jenkins-queue_not_important -n DIRAC.Jenkins.ch --wnVO=dteam --cert --certLocation=${GITHUB_WORKSPACE}/Pilot/etc/grid-security --debug
+          g_job="testintegrationworkflow${GITHUB_JOB//-/}"
+          pilotUUID="${g_job//_/}""${GITHUB_RUN_ID}"
+          pilotUUID=$(echo $pilotUUID | rev | cut -c 1-32 | rev)
+          python dirac-pilot.py --modules https://github.com/DIRACGrid/DIRAC.git:::DIRAC:::integration -M 1 -S DIRAC-Certification -N jenkins-full.cern.ch -Q jenkins-queue_not_important -n DIRAC.Jenkins.ch --wnVO=dteam --cert --certLocation=${GITHUB_WORKSPACE}/Pilot/etc/grid-security --pilotUUID="${pilotUUID}" --debug
 
 
   integration-cvmfs_matching:
@@ -236,7 +248,10 @@ jobs:
           sed -i "s#VAR_CS#https://lbcertifdirac70.cern.ch:9135/Configuration/Server#g" pilot.json
           sed -i "s#VAR_USERDN#/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=alboyer/CN=819281/CN=Alexandre Franck Boyer#g" pilot.json
           sed -i "s#VAR_USERDN_GRIDPP#${DIRACUSERDN_GRIDPP}#g" pilot.json
-          python dirac-pilot.py -M 1 -S DIRAC-Certification -N jenkins-full.cern.ch -Q jenkins-queue_not_important -n DIRAC.Jenkins.ch --preinstalledEnvPrefix=/cvmfs/dirac.egi.eu/dirac --cert --certLocation=${GITHUB_WORKSPACE}/Pilot/etc/grid-security --wnVO=${{ matrix.VO }} -o cvmfsOnly --debug
+          g_job="testintegrationworkflow${GITHUB_JOB//-/}"
+          pilotUUID="${g_job//_/}""${GITHUB_RUN_ID}"
+          pilotUUID=$(echo $pilotUUID | rev | cut -c 1-32 | rev)
+          python dirac-pilot.py -M 1 -S DIRAC-Certification -N jenkins-full.cern.ch -Q jenkins-queue_not_important -n DIRAC.Jenkins.ch --preinstalledEnvPrefix=/cvmfs/dirac.egi.eu/dirac --cert --certLocation=${GITHUB_WORKSPACE}/Pilot/etc/grid-security --wnVO=${{ matrix.VO }} -o cvmfsOnly --pilotUUID="${pilotUUID}" --debug
 
 
   ##################################
@@ -293,7 +308,10 @@ jobs:
           sed -i "s#VAR_CS#https://lbcertifdirac70.cern.ch:9135/Configuration/Server#g" pilot.json
           sed -i "s#VAR_USERDN#/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=alboyer/CN=819281/CN=Alexandre Franck Boyer#g" pilot.json
           sed -i "s#VAR_USERDN_GRIDPP#${DIRACUSERDN_GRIDPP}#g" pilot.json
-          python dirac-pilot.py -M 1 -S DIRAC-Certification -N jenkins-lhcb.cern.ch -Q jenkins-queue_not_important -n DIRAC.Jenkins.ch -e LHCb -l LHCb -E LHCbPilot --cert --certLocation=${GITHUB_WORKSPACE}/Pilot/etc/grid-security --wnVO=LHCb  -o lbRunOnly --debug
+          g_job="testintegrationworkflow${GITHUB_JOB//-/}"
+          pilotUUID="${g_job//_/}""${GITHUB_RUN_ID}"
+          pilotUUID=$(echo $pilotUUID | rev | cut -c 1-32 | rev)
+          python dirac-pilot.py -M 1 -S DIRAC-Certification -N jenkins-lhcb.cern.ch -Q jenkins-queue_not_important -n DIRAC.Jenkins.ch -e LHCb -l LHCb -E LHCbPilot --cert --certLocation=${GITHUB_WORKSPACE}/Pilot/etc/grid-security --wnVO=LHCb  -o lbRunOnly --pilotUUID="${pilotUUID}" --debug
 
   ext-lhcb_integration_dirac_installer_no_env:
     runs-on: ubuntu-latest
@@ -337,7 +355,10 @@ jobs:
           sed -i "s#VAR_CS#https://lbcertifdirac70.cern.ch:9135/Configuration/Server#g" pilot.json
           sed -i "s#VAR_USERDN#/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=alboyer/CN=819281/CN=Alexandre Franck Boyer#g" pilot.json
           sed -i "s#VAR_USERDN_GRIDPP#${DIRACUSERDN_GRIDPP}#g" pilot.json
-          python dirac-pilot.py -M 1 -S DIRAC-Certification -N ${{ matrix.ce }} -Q jenkins-queue_not_important -n DIRAC.Jenkins.ch -e LHCb -l LHCb -E LHCbPilot --preinstalledEnvPrefix=/cvmfs/lhcb.cern.ch/lhcbdirac/ --cert --certLocation=${GITHUB_WORKSPACE}/Pilot/etc/grid-security --wnVO=LHCb -o cvmfsOnly --debug
+          g_job="testintegrationworkflow${GITHUB_JOB//-/}"
+          pilotUUID="${g_job//_/}""${GITHUB_RUN_ID}"
+          pilotUUID=$(echo $pilotUUID | rev | cut -c 1-32 | rev)
+          python dirac-pilot.py -M 1 -S DIRAC-Certification -N ${{ matrix.ce }} -Q jenkins-queue_not_important -n DIRAC.Jenkins.ch -e LHCb -l LHCb -E LHCbPilot --preinstalledEnvPrefix=/cvmfs/lhcb.cern.ch/lhcbdirac/ --cert --certLocation=${GITHUB_WORKSPACE}/Pilot/etc/grid-security --wnVO=LHCb -o cvmfsOnly --pilotUUID="${pilotUUID}" --debug
 
   ext-lhcb_integration_no_env:
     runs-on: ubuntu-latest
@@ -381,4 +402,7 @@ jobs:
           sed -i "s#VAR_CS#https://lbcertifdirac70.cern.ch:9135/Configuration/Server#g" pilot.json
           sed -i "s#VAR_USERDN#/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=alboyer/CN=819281/CN=Alexandre Franck Boyer#g" pilot.json
           sed -i "s#VAR_USERDN_GRIDPP#${DIRACUSERDN_GRIDPP}#g" pilot.json
-          python dirac-pilot.py -M 1 -S DIRAC-Certification -N jenkins-lhcb.cern.ch -Q jenkins-queue_not_important -n DIRAC.Jenkins.ch -e LHCb -l LHCb -E LHCbPilot --preinstalledEnvPrefix=/cvmfs/lhcb.cern.ch/lhcbdirac/ --cert --certLocation=${GITHUB_WORKSPACE}/Pilot/etc/grid-security --wnVO=LHCb -o cvmfsOnly --debug
+          g_job="testintegrationworkflow${GITHUB_JOB//-/}"
+          pilotUUID="${g_job//_/}""${GITHUB_RUN_ID}"
+          pilotUUID=$(echo $pilotUUID | rev | cut -c 1-32 | rev)
+          python dirac-pilot.py -M 1 -S DIRAC-Certification -N jenkins-lhcb.cern.ch -Q jenkins-queue_not_important -n DIRAC.Jenkins.ch -e LHCb -l LHCb -E LHCbPilot --preinstalledEnvPrefix=/cvmfs/lhcb.cern.ch/lhcbdirac/ --cert --certLocation=${GITHUB_WORKSPACE}/Pilot/etc/grid-security --wnVO=LHCb -o cvmfsOnly --pilotUUID="${pilotUUID}" --debug

--- a/Pilot/pilotCommands.py
+++ b/Pilot/pilotCommands.py
@@ -498,7 +498,6 @@ class InstallDIRAC(CommandBase):
             # environment variables to add?
             if self.pp.userEnvVariables:
                 # User-requested environment variables (comma-separated, name and value separated by ":::")
-                # The case of dirac-install is handled in vanilla DIRAC Pilot
                 newEnvVars = dict(
                 name.split(":::", 1) for name in self.pp.userEnvVariables.replace(" ", "").split(",")
               )

--- a/Pilot/pilotCommands.py
+++ b/Pilot/pilotCommands.py
@@ -206,6 +206,7 @@ class CheckWorkerNode(CommandBase):
             )
             self.exitWithError(1)
 
+
 class InstallDIRAC(CommandBase):
     """ Source from CVMFS, or install locally
     """
@@ -525,7 +526,7 @@ class ConfigureBasics(CommandBase):
 
     It calls dirac-configure to:
 
-        * download, by default, the CAs
+        * (maybe) download the CAs
         * creates a standard or custom (defined by self.pp.localConfigFile) cfg file
           (by default 'pilot.cfg') to be used where all the pilot configuration is to be set, e.g.:
         * adds to it basic info like the version
@@ -604,37 +605,6 @@ class ConfigureBasics(CommandBase):
         if self.pp.wnVO:
             self.cfg.append('-o "/Resources/Computing/CEDefaults/VirtualOrganization=%s"' % self.pp.wnVO)
 
-    def __checkSecurityDir(self, envName, dirName):
-
-        if envName in os.environ and safe_listdir(os.environ[envName]):
-            self.log.debug(
-                "%s is set in the host environment as %s, aligning installEnv to it"
-                % (envName, os.environ[envName])
-            )               
-            self.pp.installEnv[envName] = os.environ[envName]
-        else:
-            self.log.debug("%s is not set in the host environment" % envName)
-            # try and find it
-            for candidate in self.pp.CVMFS_locations:
-                candidateDir = os.path.join(candidate,
-                                            'etc/grid-security',
-                                            dirName)
-                self.log.debug(
-                    "Candidate directory for %s is %s"
-                    % (envName, candidateDir)
-                )
-                if safe_listdir(candidateDir):
-
-                    self.log.debug("Setting %s=%s" % (envName, candidateDir))
-                    self.pp.installEnv[envName] = candidateDir
-                    os.environ[envName] = candidateDir
-                    break
-                self.log.debug("%s not found or not a directory" % candidateDir)
-
-        if envName not in self.pp.installEnv:
-            self.log.error("Could not find/set %s" % envName)
-            sys.exit(1)
-
     def _getSecurityCFG(self):
         """ Sets security-related env variables, if needed
         """
@@ -644,17 +614,8 @@ class ConfigureBasics(CommandBase):
             self.cfg.append("-o /DIRAC/Security/CertFile=%s/hostcert.pem" % self.pp.certsLocation)
             self.cfg.append("-o /DIRAC/Security/KeyFile=%s/hostkey.pem" % self.pp.certsLocation)
 
-        # If DIRAC (or its extension) is installed in CVMFS:
+        # If DIRAC (or its extension) is installed in CVMFS do not download VOMS and CAs
         if self.pp.preinstalledEnv:
-
-            self.__checkSecurityDir("X509_CERT_DIR", "certificates")
-            self.__checkSecurityDir("X509_VOMS_DIR", "vomsdir")
-            self.__checkSecurityDir("X509_VOMSES", "vomses")
-            # This is needed for the integration tests
-            self.pp.installEnv["DIRAC_VOMSES"] = self.pp.installEnv["X509_VOMSES"]
-            os.environ["DIRAC_VOMSES"] = os.environ["X509_VOMSES"]
-
-            # In any case do not download VOMS and CAs
             self.cfg.append("-DMH")
 
 

--- a/Pilot/pilotTools.py
+++ b/Pilot/pilotTools.py
@@ -1195,6 +1195,10 @@ class PilotParams(object):
         self.releaseProject = pilotOptions.get("Project", self.releaseProject)  # default from the code.
         self.log.debug("Release project: %s" % self.releaseProject)
 
+        self.CVMFS_locations = pilotOptions.get("CVMFS_locations", self.CVMFS_locations)  # default from the code.
+        self.log.debug("CVMFS locations: %s" % self.CVMFS_locations)
+
+
     def getPilotOptionsDict(self):
         """
         Get pilot option dictionary by searching paths in a certain order (commands, logging etc.).

--- a/Pilot/pilotTools.py
+++ b/Pilot/pilotTools.py
@@ -850,7 +850,7 @@ class PilotParams(object):
         self.pilotCFGFile = "pilot.json"
         self.pilotLogging = False
         self.loggerURL = None
-        self.loggerTimerInterval = 600
+        self.loggerTimerInterval = 0
         self.pilotUUID = "unknown"
         self.modules = ""  # see dirac-install "-m" option documentation
         self.userEnvVariables = ""  # see dirac-install "--userEnvVariables" option documentation

--- a/Pilot/pilotTools.py
+++ b/Pilot/pilotTools.py
@@ -51,6 +51,11 @@ try:
 except NameError:
     FileNotFoundError = OSError
 
+try:
+    IsADirectoryError  # pylint: disable=used-before-assignment
+except NameError:
+    IsADirectoryError = OSError
+
 # Timer 2.7 and < 3.3 versions issue where Timer is a function
 if sys.version_info.major == 2 or sys.version_info.major == 3 and sys.version_info.minor < 3:
     from threading import _Timer as Timer  # pylint: disable=no-name-in-module

--- a/Pilot/pilotTools.py
+++ b/Pilot/pilotTools.py
@@ -8,7 +8,6 @@ import getopt
 import imp
 import json
 import os
-import pickle
 import re
 import select
 import signal
@@ -17,7 +16,6 @@ import subprocess
 import sys
 import threading
 from datetime import datetime
-from distutils.version import LooseVersion
 from functools import partial, wraps
 from threading import RLock
 
@@ -87,15 +85,6 @@ def parseVersion(releaseVersion, useLegacyStyle):
         if pre:
             version += "a" + pre
     return version
-
-
-def printVersion(log):
-    log.info("Running %s" % " ".join(sys.argv))
-    try:
-        with open("%s.run" % sys.argv[0], "w") as fd:
-            pickle.dump(sys.argv[1:], fd)
-    except OSError:
-        pass
 
 
 def pythonPathCheck():
@@ -632,7 +621,13 @@ def sendMessage(url, pilotUUID, method, rawMessage):
 
     context = ssl.create_default_context()
     context.load_verify_locations(capath=caPath)
-    context.load_cert_chain(cert)
+    try:
+        context.load_cert_chain(cert)
+    except IsADirectoryError:
+        context.load_cert_chain(
+            os.path.join(cert, "hostcert.pem"),
+            os.path.join(cert, "hostkey.pem")
+        )
     res = urlopen(url, data, context=context)
     res.close()
 
@@ -940,6 +935,48 @@ class PilotParams(object):
         # Command line can override options from JSON
         self.__initCommandLine2()
 
+        self.__checkSecurityDir("X509_CERT_DIR", "certificates")
+        self.__checkSecurityDir("X509_VOMS_DIR", "vomsdir")
+        self.__checkSecurityDir("X509_VOMSES", "vomses")
+        # This is needed for the integration tests
+        self.installEnv["DIRAC_VOMSES"] = self.installEnv["X509_VOMSES"]
+        os.environ["DIRAC_VOMSES"] = os.environ["X509_VOMSES"]
+
+        if self.useServerCertificate:
+            self.installEnv["X509_USER_PROXY"] = self.certsLocation
+            os.environ["X509_USER_PROXY"] = self.certsLocation
+
+    def __checkSecurityDir(self, envName, dirName):
+
+        if envName in os.environ and safe_listdir(os.environ[envName]):
+            self.log.debug(
+                "%s is set in the host environment as %s, aligning installEnv to it"
+                % (envName, os.environ[envName])
+            )               
+            self.installEnv[envName] = os.environ[envName]
+        else:
+            self.log.debug("%s is not set in the host environment" % envName)
+            # try and find it
+            for candidate in self.CVMFS_locations:
+                candidateDir = os.path.join(candidate,
+                                            'etc/grid-security',
+                                            dirName)
+                self.log.debug(
+                    "Candidate directory for %s is %s"
+                    % (envName, candidateDir)
+                )
+                if safe_listdir(candidateDir):
+                    self.log.debug("Setting %s=%s" % (envName, candidateDir))
+                    self.installEnv[envName] = candidateDir
+                    os.environ[envName] = candidateDir
+                    break
+                self.log.debug("%s not found or not a directory" % candidateDir)
+
+        if envName not in self.installEnv:
+            self.log.error("Could not find/set %s" % envName)
+            sys.exit(1)
+
+
     def __initCommandLine1(self):
         """Parses and interpret options on the command line: first pass (essential things)"""
 
@@ -1185,7 +1222,7 @@ class PilotParams(object):
             except IOError as err:
                 self.log.error("Could not read a proxy, setting vo to 'unknown': ", os.strerror(err.errno))
         else:
-            self.log.error("Could not locate a proxy via X509_USER_PROXY, setting vo to 'unknown' ")
+            self.log.error("Could not locate a proxy via X509_USER_PROXY")
 
         # is there a token, and can we get a VO from the token?
         # TBD

--- a/Pilot/pilotTools.py
+++ b/Pilot/pilotTools.py
@@ -610,24 +610,29 @@ def sendMessage(url, pilotUUID, method, rawMessage):
     :param str rawMessage: a message to be sent, in JSON format
     :return: None.
     """
-    message = json.dumps((json.dumps(rawMessage), pilotUUID))
-    major, minor, micro, _, _ = sys.version_info
-    if major >= 3:
-        data = urlencode({"method": method, "args": message}).encode("utf-8")  # encode to bytes ! for python3
-    else:
-        data = urlencode({"method": method, "args": message})
     caPath = os.getenv("X509_CERT_DIR")
     cert = os.getenv("X509_USER_PROXY")
 
     context = ssl.create_default_context()
     context.load_verify_locations(capath=caPath)
+    
+    message = json.dumps((json.dumps(rawMessage), pilotUUID))
+
     try:
-        context.load_cert_chain(cert)
-    except IsADirectoryError:
+        context.load_cert_chain(cert)  # this is a proxy
+        raw_data = {"method": method, "args": message}
+    except IsADirectoryError:  # assuming it'a dir containing cert and key
         context.load_cert_chain(
             os.path.join(cert, "hostcert.pem"),
             os.path.join(cert, "hostkey.pem")
         )
+        raw_data = {"method": method, "args": message, "extraCredentials": '"hosts"'}
+    
+    if sys.version_info[0] == 3:
+        data = urlencode(raw_data).encode("utf-8")  # encode to bytes ! for python3
+    else:
+        data = urlencode(raw_data)
+
     res = urlopen(url, data, context=context)
     res.close()
 

--- a/Pilot/tests/Test_Pilot.py
+++ b/Pilot/tests/Test_Pilot.py
@@ -3,18 +3,18 @@
 
 from __future__ import absolute_import, division, print_function
 
-# pylint: disable=protected-access, missing-docstring, invalid-name, line-too-long
-
-# imports
-import unittest
 import json
-import stat
-import sys
 import os
 import shutil
+import stat
+import sys
 
-from Pilot.pilotTools import PilotParams
+# pylint: disable=protected-access, missing-docstring, invalid-name, line-too-long
+# imports
+import unittest
+
 from Pilot.pilotCommands import CheckWorkerNode, ConfigureSite, NagiosProbes
+from Pilot.pilotTools import PilotParams
 
 
 class PilotTestCase(unittest.TestCase):
@@ -39,6 +39,10 @@ class PilotTestCase(unittest.TestCase):
                 },
                 fp,
             )
+        os.environ["X509_CERT_DIR"] = os.getcwd()
+        os.environ["X509_VOMS_DIR"] = os.getcwd()
+        os.environ["X509_VOMSES"] = os.getcwd()
+        os.environ["X509_USER_PROXY"] = os.getcwd()
 
     def tearDown(self):
         for fileProd in [

--- a/Pilot/tests/Test_proxyTools.py
+++ b/Pilot/tests/Test_proxyTools.py
@@ -1,12 +1,10 @@
-from __future__ import print_function
-from __future__ import division
-from __future__ import absolute_import
+from __future__ import absolute_import, division, print_function
 
-import subprocess
-import shlex
-import sys
 import os
+import shlex
 import shutil
+import subprocess
+import sys
 import unittest
 
 ############################
@@ -49,7 +47,7 @@ class TestProxyTools(unittest.TestCase):
             data = fp.read()
         os.remove(cert)
         with self.assertRaises(Exception) as exc:
-            vo = getVO(data)
+            getVO(data)
 
         self.assertEqual(str(exc.exception), msg)
 
@@ -58,7 +56,7 @@ class TestProxyTools(unittest.TestCase):
         msg = "command not found: openssl"
         popenMock.side_effect = OSError(msg)
         with self.assertRaises(OSError) as exc:
-            res = getVO(data)
+            getVO(data)
         self.assertEqual(str(exc.exception), msg)
 
     @patch("Pilot.proxyTools.Popen")
@@ -70,7 +68,7 @@ class TestProxyTools(unittest.TestCase):
         msg = "Error when invoking openssl asn1parse"
 
         with self.assertRaises(Exception) as exc:
-            res = parseASN1("Anything")
+            parseASN1("Anything")
 
         self.assertEqual(str(exc.exception), msg)
 
@@ -80,7 +78,7 @@ class TestProxyTools(unittest.TestCase):
         msg = "command not found: openssl"
         popenMock.side_effect = OSError(msg)
         with self.assertRaises(OSError) as exc:
-            res = parseASN1("Anything")
+            parseASN1("Anything")
         self.assertEqual(str(exc.exception), msg)
 
     def test_createFakeProxy(self):

--- a/Pilot/tests/Test_simplePilotLogger.py
+++ b/Pilot/tests/Test_simplePilotLogger.py
@@ -34,6 +34,10 @@ class TestPilotParams(unittest.TestCase):
             "tests/pilot.json",
         ]
 
+        os.environ["X509_CERT_DIR"] = os.getcwd()
+        os.environ["X509_VOMS_DIR"] = os.getcwd()
+        os.environ["X509_VOMSES"] = os.getcwd()
+        os.environ["X509_USER_PROXY"] = os.getcwd()
         pp = PilotParams()
 
         argvmock.__getitem__.assert_called()
@@ -89,6 +93,10 @@ class TestPilotParams(unittest.TestCase):
             "/%s/Pilot" % vo
         ]
         mockPaths.return_value = paths
+        os.environ["X509_CERT_DIR"] = os.getcwd()
+        os.environ["X509_VOMS_DIR"] = os.getcwd()
+        os.environ["X509_VOMSES"] = os.getcwd()
+        os.environ["X509_USER_PROXY"] = os.getcwd()
         pp = PilotParams()
         lTESTcommands = "CheckWorkerNode, InstallDIRAC, ConfigureBasics, RegisterPilot, CheckCECapabilities, CheckWNCapabilities, ConfigureSite, ConfigureArchitecture, ConfigureCPURequirements"
 
@@ -112,6 +120,9 @@ class TestCommandBase(unittest.TestCase):
         # These temporary files, opened in a binary mode, will act as standard stream pipes for `Popen`
         self.stdout_mock = tempfile.NamedTemporaryFile(mode="rb+", delete=False)
         self.stderr_mock = tempfile.NamedTemporaryFile(mode="rb+", delete=False)
+        os.environ["X509_CERT_DIR"] = '/some/thing/'
+        os.environ["X509_USER_PROXY"] = '/some/thing'
+
 
     def tearDown(self):
         # At the end of the test, we'll close and remove the created files
@@ -122,7 +133,10 @@ class TestCommandBase(unittest.TestCase):
     @patch(("sys.argv"))
     @patch("subprocess.Popen")
     def test_executeAndGetOutput(self, popenMock, argvmock):
-
+        os.environ["X509_CERT_DIR"] = os.getcwd()
+        os.environ["X509_VOMS_DIR"] = os.getcwd()
+        os.environ["X509_VOMSES"] = os.getcwd()
+        os.environ["X509_USER_PROXY"] = os.getcwd()
         argvmock.__getitem__.return_value = [
             "-d",
             "-g",

--- a/Pilot/tests/Test_simplePilotLogger.py
+++ b/Pilot/tests/Test_simplePilotLogger.py
@@ -95,7 +95,7 @@ class TestPilotParams(unittest.TestCase):
         pp.gridCEType = "TEST"
 
         res = pp.getPilotOptionsDict()
-        logURL = "https://lbvobox70.cern.ch:8443/WorkloadManagement/TornadoPilotLogging"
+        logURL = "https://lbcertifdirac70.cern.ch:8443/WorkloadManagement/TornadoPilotLogging"
         self.assertEqual(res.get("RemoteLoggerURL"), logURL)
         self.assertEqual(pp.loggerURL, logURL)
         self.assertEqual(res.get("RemoteLogging"), "False")

--- a/tests/CI/pilot_newSchema.json
+++ b/tests/CI/pilot_newSchema.json
@@ -123,7 +123,7 @@
 			"pilotRepo": "https://github.com/should_not_matter/Pilot.git",
 			"GenericPilotGroup": "dteam_pilot",
 			"GenericPilotDN": "VAR_USERDN",
-			"RemoteLogging": "False",
+			"RemoteLogging": "True",
 			"RemoteLoggerURL": "https://lbvobox70.cern.ch:8443/WorkloadManagement/TornadoPilotLogging",
 			"PilotLogLevel": "DEBUG"
 		}

--- a/tests/CI/pilot_newSchema.json
+++ b/tests/CI/pilot_newSchema.json
@@ -101,13 +101,13 @@
 			"Version": "VAR_DIRAC_VERSION",
 			"Extensions": "None",
 			"CheckVersion": "False",
-			"pilotFileServer": "lbvobox70.cern.ch:8443",
+			"pilotFileServer": "lbcertifdirac70.cern.ch:8443",
 			"pilotRepoBranch": "does_not_matter",
 			"pilotRepo": "https://github.com/does_not_matter/Pilot.git",
 			"GenericPilotGroup": "gridpp_pilot",
 			"GenericPilotDN": "VAR_USERDN_GRIDPP",
 			"RemoteLogging": "False",
-			"RemoteLoggerURL": "https://lbvobox70.cern.ch:8443/WorkloadManagement/TornadoPilotLogging",
+			"RemoteLoggerURL": "https://lbcertifdirac70.cern.ch:8443/WorkloadManagement/TornadoPilotLogging",
 			"UploadSE": "UKI-LT2-IC-HEP-disk",
 			"UploadPath": "/gridpp/pilotlogs/",
 			"LoggingShifterName": "GridPPLogManager",
@@ -118,13 +118,13 @@
 		"Pilot": {
 			"Version": "VAR_DIRAC_VERSION",
 			"CheckVersion": "True",
-			"pilotFileServer": "lbvobox70.cern.ch:8443",
+			"pilotFileServer": "lbcertifdirac70.cern.ch:8443",
 			"pilotRepoBranch": "should_not_matter",
 			"pilotRepo": "https://github.com/should_not_matter/Pilot.git",
 			"GenericPilotGroup": "dteam_pilot",
 			"GenericPilotDN": "VAR_USERDN",
 			"RemoteLogging": "True",
-			"RemoteLoggerURL": "https://lbvobox70.cern.ch:8443/WorkloadManagement/TornadoPilotLogging",
+			"RemoteLoggerURL": "https://lbcertifdirac70.cern.ch:8443/WorkloadManagement/TornadoPilotLogging",
 			"PilotLogLevel": "DEBUG"
 		}
 	},
@@ -132,11 +132,11 @@
 		"Pilot": {
 			"Version": "VAR_DIRAC_VERSION",
 			"CheckVersion": "True",
-			"pilotFileServer": "lbvobox70.cern.ch:8443",
+			"pilotFileServer": "lbcertifdirac70.cern.ch:8443",
 			"GenericPilotGroup": "dteam_pilot",
 			"GenericPilotDN": "VAR_USERDN",
 			"RemoteLogging": "False",
-			"RemoteLoggerURL": "https://lbvobox70.cern.ch:8443/WorkloadManagement/TornadoPilotLogging",
+			"RemoteLoggerURL": "https://lbcertifdirac70.cern.ch:8443/WorkloadManagement/TornadoPilotLogging",
 			"PilotLogLevel": "DEBUG"
 		}
 	},

--- a/tests/CI/pilot_newSchema.json
+++ b/tests/CI/pilot_newSchema.json
@@ -110,6 +110,7 @@
 			"RemoteLoggerURL": "https://lbcertifdirac70.cern.ch:8443/WorkloadManagement/TornadoPilotLogging",
 			"UploadSE": "UKI-LT2-IC-HEP-disk",
 			"UploadPath": "/gridpp/pilotlogs/",
+			"RemoteLoggerTimerInterval": 0,
 			"LoggingShifterName": "GridPPLogManager",
 			"PilotLogLevel": "DEBUG"
 		}
@@ -125,6 +126,7 @@
 			"GenericPilotDN": "VAR_USERDN",
 			"RemoteLogging": "True",
 			"RemoteLoggerURL": "https://lbcertifdirac70.cern.ch:8443/WorkloadManagement/TornadoPilotLogging",
+			"RemoteLoggerTimerInterval": 0,
 			"PilotLogLevel": "DEBUG"
 		}
 	},

--- a/tests/CI/pilot_newSchema.json
+++ b/tests/CI/pilot_newSchema.json
@@ -118,7 +118,7 @@
 	"dteam": {
 		"Pilot": {
 			"Version": "VAR_DIRAC_VERSION",
-			"CheckVersion": "True",
+			"CheckVersion": "False",
 			"pilotFileServer": "lbcertifdirac70.cern.ch:8443",
 			"pilotRepoBranch": "should_not_matter",
 			"pilotRepo": "https://github.com/should_not_matter/Pilot.git",
@@ -133,12 +133,12 @@
 	"LHCb": {
 		"Pilot": {
 			"Version": "VAR_DIRAC_VERSION",
-			"CheckVersion": "True",
+			"CheckVersion": "False",
 			"pilotFileServer": "lbcertifdirac70.cern.ch:8443",
 			"GenericPilotGroup": "dteam_pilot",
 			"GenericPilotDN": "VAR_USERDN",
 			"RemoteLogging": "False",
-			"RemoteLoggerURL": "https://lbcertifdirac70.cern.ch:8443/WorkloadManagement/TornadoPilotLogging",
+			"RemoteLoggerTimerInterval": 0,
 			"PilotLogLevel": "DEBUG"
 		}
 	},

--- a/tests/CI/pilot_newSchema.json
+++ b/tests/CI/pilot_newSchema.json
@@ -112,7 +112,8 @@
 			"UploadPath": "/gridpp/pilotlogs/",
 			"RemoteLoggerTimerInterval": 0,
 			"LoggingShifterName": "GridPPLogManager",
-			"PilotLogLevel": "DEBUG"
+			"PilotLogLevel": "DEBUG",
+			"CVMFS_locations": ["/cvmfs/grid.cern.ch", "/cvmfs/dirac.egi.eu"]
 		}
 	},
 	"dteam": {
@@ -127,7 +128,8 @@
 			"RemoteLogging": "True",
 			"RemoteLoggerURL": "https://lbcertifdirac70.cern.ch:8443/WorkloadManagement/TornadoPilotLogging",
 			"RemoteLoggerTimerInterval": 0,
-			"PilotLogLevel": "DEBUG"
+			"PilotLogLevel": "DEBUG",
+			"CVMFS_locations": ["/cvmfs/grid.cern.ch", "/cvmfs/dirac.egi.eu"]
 		}
 	},
 	"LHCb": {
@@ -139,7 +141,8 @@
 			"GenericPilotDN": "VAR_USERDN",
 			"RemoteLogging": "False",
 			"RemoteLoggerTimerInterval": 0,
-			"PilotLogLevel": "DEBUG"
+			"PilotLogLevel": "DEBUG",
+			"CVMFS_locations": ["/cvmfs/lhcb.cern.ch/", "$VO_LHCB_SW_DIR"]
 		}
 	},
 	"ConfigurationServers": [

